source/Calamari/Kubernetes/ResourceStatus/Resources/Pod.cs

```diff
@@ -1,36 +1,121 @@
+using System.Linq;
 using Newtonsoft.Json.Linq;
 
 namespace Calamari.Kubernetes.ResourceStatus.Resources
 {
     public class Pod : Resource
     {
-        public string Phase { get; }
+        public string Status { get; }
         public override ResourceStatus ResourceStatus { get; }
     
         public Pod(JObject json) : base(json)
         {
-            Phase = Field("$.status.phase");
-            
-            // TODO implement this
-            switch (Phase)
+            var phase = Field("$.status.phase");
+            var initContainerStatuses = data
+                .SelectToken("$.status.initContainerStatuses")
+                ?.ToObject<ContainerStatus[]>() ?? new ContainerStatus[] { };
+            var containerStatuses = data
+                .SelectToken("$.status.containerStatuses")
+                ?.ToObject<ContainerStatus[]>() ?? new ContainerStatus[] { };
+
+            (Status, ResourceStatus) = GetStatus(phase, initContainerStatuses, containerStatuses);
+        }
+    
+        public override bool HasUpdate(Resource lastStatus)
+        {
+            var last = CastOrThrow<Pod>(lastStatus);
+            return last.ResourceStatus != ResourceStatus || last.Status != Status;
+        }
+
+        private (string, ResourceStatus) GetStatus(
+            string phase, 
+            ContainerStatus[] initContainerStatuses,
+            ContainerStatus[] containerStatuses)
+        {
+            switch (phase)
             {
-                case"Succeeded": 
-                case "Running":
-                    ResourceStatus = ResourceStatus.Successful;
-                    break;
                 case "Pending":
-                    ResourceStatus = ResourceStatus.InProgress;
-                    break;
+                    if (!initContainerStatuses.Any() && !containerStatuses.Any())
+                    {
+                        return ("Pending", ResourceStatus.InProgress);
```

Comment:
Maybe this one should return the "queued" status?

Reply:
We don't have the state for "queued" defined for a k8s resource and we don't know what's the definition of "queued" or how to differentiate it from "InProgress". I imagine you may think "queued" is appropriate because it is already one of the states we use in frontend as you showed me today. But in my opinion, using the frontend display status to drive the business logic would not be a proper choice. Also I think the frontend "queued" status may have other implications in other contexts, which may not be proper in the k8s status check context.

Reply:
no worries, sounds ok